### PR TITLE
Adds database to the app

### PR DESCRIPTION
**Pull request by corps**
Related: https://github.com/getsentry/ops/pull/9351

@hubertdeng123 Is there a safe way to ensure that this + the new gocd pipeline step actually works without firing off?  I wonder how hard a canary for seer would be.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,39 +14,42 @@ help:
...
 	  -v $(tmpdir):/sentry-data-schemas:ro \
 	  -v $$(pwd)/src/:/src:ro \
 	  tufin/oasdiff breaking /sentry-data-schemas/seer/seer_api.json /src/seer/schemas/seer_api.json
+
+.PHONY: migration
+migration: .env
+	docker-compose run app flask db migrate -m 'Migration'
+
+.env:
+	cp .env.example .env
```
Comment (jennmueng) [inline Makefile:55]: .env.example would be empty unless you set the secrets, do we want to be copying the file without values set over?

File: src/migrations/env.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+logger = logging.getLogger("alembic.env")
+
+
+def get_engine():
+    try:
+        # this works with Flask-SQLAlchemy<3 and Alchemical
+        return current_app.extensions["migrate"].db.get_engine()
+    except (TypeError, AttributeError):
+        # this works with Flask-SQLAlchemy>=3
+        return current_app.extensions["migrate"].db.engine
```
Comment (jennmueng) [inline src/migrations/env.py:24]: don't we have 3.1.1 set in the requirements.txt?

**Review by jennmueng — APPROVED** [review]
some comments, lgtm otherwise! Thanks for setting this up @corps!

File: Makefile (hunk 1 above)
Comment (corps) [reply Makefile:55]: For CI, for instance, yes.  `.env.example` is a good place to have 'defaults' given that CI will depend on the docker-compose file for running moving forward.

That said, I'm curious about options for letting users know that they should be editing this file robustly.  Like a big echo?  Mainly I want to keep CI simple (and the easy case of running tests simple).

File: src/migrations/env.py (hunk 2 above)
Comment (corps) [reply src/migrations/env.py:24]: We do, this file is actually auto generated from the `flask db` command, so I can likely clean this stuff up.

File: Makefile (hunk 1 above)
Comment (jennmueng) [reply Makefile:55]: Most of the secrets are fine to get away without in CI so it should work even with them not set in this file.

We should probably add warnings when running seer without required env variables which should be more robust than just an echo

Comment (corps) [reply Makefile:55]: The issue isn't about running seer, it's about the `docker-compose.yml` configuration that requires the `.env` file to exist, regardless of wether it is used (which it isn't in CI, but the file still must exist).

**Pull request merged**